### PR TITLE
Start using the `block.json` metadata file for `Handpicked products`

### PR DESCRIPTION
**Pull request by albarin**
The goal of this PR is to use the `block.json` metadata file to register the `Handpicked products` block.

### Testing

How to test the changes in this Pull Request:
1. Create a page and add a `Handpicked products` block.
2. Save the page and check it renders correctly the default config in the frontend.
3. Edit the page again and make some changes to the block (hiding some content, changing columns number, ordering, etc.).
4. Save the page and check it renders properly with the new config.

### CODE REVIEW

File: assets/js/blocks/handpicked-products/block.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
+    "name": "woocommerce/handpicked-products",
+    "version": "1.0.0",
+    "title": "Hand-picked Products",
+    "category": "woocommerce",
+    "keywords": [
+        "Handpicked Products",
+        "WooCommerce"
+    ],
+    "description": "Display a selection of hand-picked products in a grid.",
```
Comment (tomasztunik) [inline assets/js/blocks/handpicked-products/block.json:9]: here and in couple of other places in old pattern we used i18n `__` methods to provide translated strings. With `block.json` we don't have the ability to provide that here. Have you looked into how we should go about this?

File: assets/js/blocks/handpicked-products/block.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            "type": "boolean",
+            "default": true
+        },
+        "contentVisibility": {
+            "type": "object",
+            "properties": {
+                "image": {
+                    "type": "boolean",
+                    "image": true
+                },
```
Comment (tomasztunik) [inline assets/js/blocks/handpicked-products/block.json:35]: great insight about the typed configuration and better data type handling!

File: assets/js/blocks/handpicked-products/block.json (hunk 1 above)
Comment (tomasztunik) [reply assets/js/blocks/handpicked-products/block.json:9]: Would be nice if it was added as a translatable string automatically

Comment (albarin) [reply assets/js/blocks/handpicked-products/block.json:9]: Yes, if you check and the end of the file there's this property `"textdomain": "woo-gutenberg-products-block"` which is used to load translations (https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#text-domains). It has the same key it was using before in the `__` function.

Comment (tomasztunik) [reply assets/js/blocks/handpicked-products/block.json:9]: ah, perfect!

File: assets/js/blocks/handpicked-products/index.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			type: 'number',
+			default: getSetting( 'default_columns', 3 ),
+		},
+	},
+
+	/**
+	 * Renders and manages the block.
+	 *
+	 * @param {Object} props Props to pass to block.
+	 */
```
Comment (tomasztunik) [inline assets/js/blocks/handpicked-products/index.tsx]: I know it was carried over but should we remove it? I believe the already present typescript hints provide more information and value

File: assets/js/blocks/handpicked-products/block.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
+{
+    "name": "woocommerce/handpicked-products",
+    "version": "1.0.0",
```
Comment (tomasztunik) [inline assets/js/blocks/handpicked-products/block.json]: looking at blocks in Gutenberg I think we should use `apiVersion: 2` here and also if we include a reference to block.json schema we'll have autocomplete and hints on some of the fields! 

All blocks in Gutenberg start with $schema and apiVersion followed by name - we could follow this pattern.

```suggestion
    "$schema": "https://schemas.wp.org/trunk/block.json",
    "apiVersion": 2,
    "name": "woocommerce/handpicked-products",
```

Comment (tomasztunik) [reply assets/js/blocks/handpicked-products/block.json]: we'd benefit from having this included everywhere where we moved to block.json so far I think

Comment (albarin) [reply assets/js/blocks/handpicked-products/block.json]: The `apiVersion` is already at the end of the json file, not sure if the `version` number is correct thought 🤔 
I'll include the `$schema`.

Comment (tomasztunik) [reply assets/js/blocks/handpicked-products/block.json]: ah, missed it sorry, ah, so the version is the version of the block? what is it?

Comment (tomasztunik) [reply assets/js/blocks/handpicked-products/block.json]: got the answer from schema :) 

> The current version number of the block, such as 1.0 or 1.0.3. It’s similar to how plugins are versioned. This field might be used with block assets to control cache invalidation, and when the block author omits it, then the installed version of WordPress is used instead.

**Review by tomasztunik — APPROVED** [review]
Looks good! 

Things I've noticed before that make me think is if we should provide standard Edit and save components/methods that we re-define multiple times, also the icon is a repeated pattern.

We could have an edit factory function that would take a block and return the default implementation that adds the wrapper and passes the `blockProps`.

or we could have a more generic factory function that would return object with defaults that we would spread when passing block configuration. It could take things like Block and icon references and return default edit, save and icon properties so that all we'd need to provide assuming there was nothing custom would be the attributes.

but not the scope of this PR just sharing thoughts.

**Pull request merged**